### PR TITLE
User roles 22 part2

### PR DESCRIPTION
**Pull request by Calacorm**
Add capability to Edit Roles

- New Endpoint @PostMapping("/user/changeRole/{id}")
- -  Example body:  { "role":"PAID_USER"}
- New Endpoint @PutMapping("/user/upgrade/{id}")
- - see javascript in view_users.html for example on how to use from js side.
- - Note no body required, this activates the upgrade/toggle capability of a role, defined in the enumerator RoleType
- - Each Role enumeration has an index that refers to a role it upgrades to, a utility method facilitates change to a user's role if the current role they have should be upgradeable.
- Added BOOTSTRAP.css lib file

How To Use:
The Main_Controller has a test method that makes an admin and a user, this procs once on entry to '/'
Login as admin:admin and click "view users"
you can even go log out and register some new users and go back and see them on the table list
The button on the table of users will toggle a user role from free to paid and back again. a no_role user will go to free, and admin will not change. The change updates every second.

Testing:
All done by hand to validate functionality
Will add Unit Tests in with resolution to issue #77 and branch User_Roles_22_part3

### CODE REVIEW

**Comment by Calacorm** [comment]
Added an update to add dynamic table support. Plan to use this tech to make a table we can edit user roles on for part of the admin UI interface

**Comment by Calacorm** [comment]
The table no shows relevant data.
The users are generated from a json file so that needs to be hooked into the JPA repository functions later on.
The table shows the users and has an edit button. THe button is not hooked up yet
Seems to be working, just need to hook up the button and hook into JPArepository instead of the json file and then liink the page to a demo button or something once you log in. And Ill have a minimum viable feature addition

**Comment by Calacorm** [comment]
Going to do polish and add tests and make sure CI passes now

File: src/main/resources/templates/Registration.html [hunk 1]
```diff
@@ -3,262 +3,43 @@
 <head>
```
Comment (Calacorm) [inline src/main/resources/templates/Registration.html:3]: Massive tool for doing dynamic tables, very cool!
https://datatables.net/

File: src/main/java/SYSC6/Project/user/User.java [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
 import javax.persistence.*;
 import java.io.Serializable;
 
 @Entity
 @Table(name = "Users")
+@Getter
+@Setter
```
Comment (Calacorm) [inline src/main/java/SYSC6/Project/user/User.java:13]: Lombok does all the boring getter setter bs for us! clean class

File: src/main/java/SYSC6/Project/RoleType.java [hunk 3]
```diff
@@ -1,8 +1,43 @@
...
+
 public enum RoleType {
-    NO_ROLE,
-    FREE_USER,
-    PAID_USER,
-    ADMIN
+    NO_ROLE (1, "no_role"),
+    FREE_USER (2, "free_user"),
+    PAID_USER (1, "paid_user"),
+    ADMIN (3, "admin");
```
Comment (Calacorm) [inline src/main/java/SYSC6/Project/RoleType.java:11]: The index parameter here is used to indicate the ordinal of the enum this role may upgrade to.
Its a bit silly, and we have an endpoint where we can just specify as well, but I implemented this system regardless

File: src/main/java/SYSC6/Project/WebConfig.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+@Configuration
+@EnableWebMvc
+@ComponentScan
+public class WebConfig implements WebMvcConfigurer {
+
+    @Override
+    public void addResourceHandlers(ResourceHandlerRegistry registry) {
+        registry.addResourceHandler("/css/**", "/Javascript/**", "/Images/**", "datatable/**", "jquery/**")
+                .addResourceLocations("classpath:/css/", "classpath:/Javascript/", "classpath:/Images/", "classpath:/Datatable/", "classpath:/jquery/");
```
Comment (Calacorm) [inline src/main/java/SYSC6/Project/WebConfig.java:15]: need an import in the html files? make sure to update this file!

File: pom.xml [hunk 5]
```diff
@@ -1,61 +1,92 @@
...
-				<groupId>org.thymeleaf.extras</groupId>
-				<artifactId>thymeleaf-extras-springsecurity5</artifactId>
-			</plugin>
 			<plugin>
 				<groupId>org.springframework.boot</groupId>
-                		<artifactId>spring-boot-maven-plugin</artifactId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
 			</plugin>
 		</plugins>
 	</build>
```
Comment (Calacorm) [inline pom.xml:92]: Going to review this, some things I added I do not need since I added the files in manually to our resources folder

**Review by NicolasDuciaume — DISMISSED** [review]
Looks good and functions as intended yet need to switch links to heroku for functioning on their system.

**Review by cadsilva — APPROVED** [review]

**Review by nazifatanzim — CHANGES_REQUESTED** [review]
Code looks good, but please add integration tests.

**Comment by Calacorm** [comment]
Heroku links set

**Comment by Calacorm** [comment]
Going to leave it in localhost mode for now until we need to show on heroku

**Review by NicolasDuciaume — APPROVED** [review]
Looks good, works as intended

**Pull request merged**